HPS/Heartland/Model/Payment.php

```diff
@@ -310,8 +302,9 @@ function validate()
             $errorMsg = __('This credit card type is not allowed for this payment method.');
         }*/
         // \HPS\Heartland\Model\Payment::getToken
-        if (!$this->getToken(new \HpsTokenData)) {
-            $errorMsg = __('No valid token.');
+        $suToken = $this->getToken(new \HpsTokenData);       
+        if(empty($suToken->tokenValue)){
```

Comment:
I haven't looked at `getToken`, but is there a change it can return an empty value? If so, we'll want to check `$suToken` before checking `$suToken->tokenValue`.